pennylane/templates/subroutines/trotter.py

```diff
@@ -0,0 +1,216 @@
+    in the hamiltonian. The symmetrized products of order :math: `m \in [1, 2, 4, ..., 2k] | k \in \mathbb{N}`
+    are given by:
+
+    .. math::
+
+        \begin{align}
+            S_{m=1}(t) &= \Pi_{j=0}^{N} \ exp(i t O_{j}) \\
+            S_{m=2}(t) &= \Pi_{j=0}^{N} \ exp(i \frac{t}{2} O_{j}) \cdot \Pi_{j=N}^{0} \ exp(i \frac{t}{2} O_{j}) \\
+            &\vdots
+            S_{m=2k}(t) &= S_{2k-2}(p_{2k}t)^{2} \cdot S_{2k-2}((1-4p_{2k})t) \cdot S_{2k-2}(p_{2k}t)^{2}
+        \end{align}
+
+    Where the coefficient is :math:`p_{2k} = \frac{1}{4 - \sqrt[2k - 1]{4}}`.
+
+    The :math:`2k`th order, :math:`n`-step Suzuki-Trotter approximation is then defined as:
+
+    .. math:: exp(iHt) \approx (S_{2k}(\frac{t}{n}))^{n}
+
+    Args:
+        hamiltonian (Union[~.Hamiltonian, ~.Sum]):
+
+    Keyword Args:
+        n (int): An integer representing the number of Trotter steps to perform.
+        order (int): An integer representing the order of the approximation (must be 1 or even).
+        check_hermitian (bool): A flag to enable the validation check to ensure this is a valid unitary operator.
+
+    Raises:
+        TypeError: The 'hamiltonian' is not of type ~.Hamiltonian, or ~.Sum.
+        ValueError: One or more of the terms in 'hamiltonian' are not Hermitian. 
+        ValueError: The 'order' is not one or a positive even integer.
+
+    **Example**
+
+    .. code-block:: python3
+
+        coeffs = [0.25, 0.75]
+        ops = [qml.PauliX(0), qml.PauliZ(0)]
+        H = qml.dot(coeffs, ops)
+
+        dev = qml.device("default.qubit", wires=2)
+        @qml.qnode(dev)
+        def my_circ():
+            # Prepare some state
+            qml.Hadamard(0)
+
+            # Evolve according to H
+            qml.TrotterProduct(H, time=2.4, order=2)
+
+            # Measure some quantity
+            return qml.state()
+
+    >>> my_circ()
+    [-0.13259524+0.59790098j  0.        +0.j         -0.13259524-0.77932754j  0.        +0.j        ]
+
+    .. details::
+        :title: Usage Details
+
+        We can also compute the gradient with respect to the coefficients of the hamiltonian and the
+        evolution time: 
+
+        .. code-block:: python3
+
+            @qml.qnode(dev)
+            def my_circ(c1, c2, time): 
+                # Prepare H: 
+                H = qml.dot([c1, c2], [qml.PauliX(0), qml.PauliZ(1)])
+                    
+                # Prepare some state
+                qml.Hadamard(0)
+                
+                # Evolve according to H
+                qml.TrotterProduct(H, time, order=2)
+                
+                # Measure some quantity
+                return qml.expval(qml.PauliZ(0) @ qml.PauliZ(1))
+
+        >>> args = qnp.array([1.23, 4.5, 0.1])
+        >>> qml.grad(my_circ)(*tuple(args))
+        (tensor(0.00961064, requires_grad=True), tensor(-0.12338274, requires_grad=True), tensor(-5.43401259, requires_grad=True))
+    """
+
```

Comment:
is not needed `num_params` or `num_wires` anymore?

Reply:
Its not required, but I can add it if needed.

Reply:
Its not needed, the default values for both match what we expect 👍🏼